pkg/alexa/gen/model.go

```
@@ -0,0 +1,266 @@
+	registry         l10n.LocaleRegistry
+	name             string
+	typeName         string
+	samplesName      string
+	withConfirmation bool
+	withElicitation  bool
+}
+
+func NewModelSlotBuilder(name string) *ModelSlotBuilder {
+	return &ModelSlotBuilder{name: name}
+}
+
+func (s *ModelSlotBuilder) WithLocaleRegistry(registry l10n.LocaleRegistry) *ModelSlotBuilder {
+	s.registry = registry
+	return s
+}
+
+func (s *ModelSlotBuilder) WithType(name string) *ModelSlotBuilder {
+	s.typeName = name
+	return s
+}
+
+func (s *ModelSlotBuilder) WithSamples(samplesName string) *ModelSlotBuilder {
+	s.samplesName = samplesName
+	return s
+}
+
+func (s *ModelSlotBuilder) WithElicitationPrompt() *ModelSlotBuilder {
+	s.withElicitation = true
+	return s
+}
+
+func (s *ModelSlotBuilder) BuildIntentSlot(locale string) alexa.ModelSlot {
+	l, _ := s.registry.Resolve(locale)
+	ms := alexa.ModelSlot{
+		Name: s.name,
+		Type: s.typeName,
+	}
+	if s.samplesName != "" {
+		ms.Samples = l.GetAll(s.samplesName)
+	}
+	return ms
+}
+
+func (s *ModelSlotBuilder) BuildDialogSlot(locale string) alexa.DialogIntentSlot {
+	ds := alexa.DialogIntentSlot{
+		Name:         s.name,
+		Type:         s.typeName,
+		Confirmation: s.withConfirmation,
+		Elicitation:  s.withElicitation,
+	}
+	return ds
+}
+
+/////////////////////////////////////////////
+
+// ModelTypeBuilder
+type ModelTypeBuilder struct {
+	registry   l10n.LocaleRegistry
+	name       string
+	valuesName string
+}
+
+func NewModelTypeBuilder(name string) *ModelTypeBuilder {
+	return &ModelTypeBuilder{name: name}
+}
+
+func (t *ModelTypeBuilder) WithLocaleRegistry(registry l10n.LocaleRegistry) *ModelTypeBuilder {
+	t.registry = registry
+	return t
+}
+
+func (t *ModelTypeBuilder) WithValuesName(valuesName string) *ModelTypeBuilder {
+	t.valuesName = valuesName
+	return t
+}
+
+func (t *ModelTypeBuilder) WithLocaleValues(locale string, values []string) *ModelTypeBuilder {
+	loc, _ := t.registry.Resolve(locale)
```

Dont ignore the error